File: sliding_window/643_max_avg_subarr_1.py
def find_max_average( nums: list[int], k: int) -> float:
    begin = 0
    window_state = 0
    result = float("-inf")

    for end in range(len(nums)):
        window_size = end - begin + 1
        window_state += nums[end]
        if window_size == k:
            result = max(result, window_state)
            window_state -= nums[begin]
            begin += 1

    return result / k

File: sliding_window/643_max_avg_subarr_1.go
import "math"

func findMaxAverage(nums []int, k int) float64 {
	begin := 0
	windowState := 0.0
	result := math.Inf(-1) // float64

	for end := range len(nums) {
		windowState += float64(nums[end])
		windowSize := end - begin + 1
		if windowSize == k {
			result = max(result, windowState)
			windowState -= float64(nums[begin])
			begin++
		}
	}

	return result / float64(k)
}